conan/tools/microsoft/subsystems.py

```diff
@@ -0,0 +1,149 @@
+import os
+import platform
+import re
+import subprocess
+
+from conan.tools.env import Environment
+from conan.tools.env.environment import environment_wrap_command
+from conans.errors import ConanException
+
+MSYS2 = 'msys2'
+MSYS = 'msys'
+CYGWIN = 'cygwin'
+WSL = 'wsl'  # Windows Subsystem for Linux
+SFU = 'sfu'  # Windows Services for UNIX
+
+
+def run_in_windows_shell(conanfile, command, cwd=None, env=None):
+    """ Will run a unix command inside a bash terminal It requires to have MSYS2, CYGWIN, or WSL"""
+    if env:
+        # Passing env invalidates the conanfile.environment_scripts
```

Comment:
Maybe call it ``env_scripts``, at first sight I thought it was requiring an environment or an env-var dict